app/code/Transoft/Blog/etc/di.xml

```diff
@@ -0,0 +1,35 @@
+<?xml version="1.0"?>
+<config xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" xsi:noNamespaceSchemaLocation="../../../../../lib/internal/Magento/Framework/ObjectManager/etc/config.xsd">
+    <type name="Magento\Framework\View\Element\UiComponent\DataProvider\CollectionFactory">
+        <arguments>
+            <argument name="collections" xsi:type="array">
+                <item name="transoft_blog_post_listing_data_source" xsi:type="string">Transoft\Blog\Model\ResourceModel\Post\Grid\Collection</item>
+            </argument>
+        </arguments>
+    </type>
+    <type name="Transoft\Blog\Model\ResourceModel\Post\Grid\Collection">
+        <arguments>
+            <argument name="mainTable" xsi:type="string">transoft_blog</argument>
+            <argument name="eventPrefix" xsi:type="string">transoft_blog_post_collection</argument>
+            <argument name="eventObject" xsi:type="string">post_grid_collection</argument>
+            <argument name="resourceModel" xsi:type="string">Transoft\Blog\Model\ResourceModel\Post</argument>
+        </arguments>
+    </type>
+    <virtualType name="BlogImageUploader" type="Magento\Catalog\Model\ImageUploader">
+        <arguments>
+            <argument name="baseTmpPath" xsi:type="string">index/upload/tmp/image</argument>
+            <argument name="basePath" xsi:type="string">index/upload/image</argument>
+            <argument name="allowedExtensions" xsi:type="array">
+                <item name="jpg" xsi:type="string">jpg</item>
+                <item name="jpeg" xsi:type="string">jpeg</item>
+                <item name="gif" xsi:type="string">gif</item>
+                <item name="png" xsi:type="string">png</item>
+            </argument>
```

Comment:
This is not necessary. Magento\Catalog\Model\ImageUploader already contains these changes

Reply:
fixed